pkg/velaql/providers/query/tree.go

```diff
@@ -393,7 +396,10 @@ func listItemByRule(clusterCTX context.Context, k8sClient client.Client, resourc
 	return itemList.Items, nil
 }
 
-func iteratorChildResources(ctx context.Context, cluster string, k8sClient client.Client, parentResource types.ResourceTreeNode) ([]types.ResourceTreeNode, error) {
+func iteratorChildResources(ctx context.Context, cluster string, k8sClient client.Client, parentResource types.ResourceTreeNode, depth int) ([]types.ResourceTreeNode, error) {
+	if depth == maxDepth {
```

Comment:
```suggestion
	if depth > maxDepth {
```

Shall we return some error or logs for reminder?

Reply:
I think we cannot generate any error, an error will break whole return chain, the result of this velaql will return error.